python/lsst/pipe/base/graph/_loadHelpers.py

```diff
@@ -0,0 +1,310 @@
+            if nodeTask not in loadedTaskDef:
+                # Get the byte ranges corresponding to this taskDef
+                start, stop = self.taskDefMap[nodeTask]
+                start += self.headerSize
+                stop += self.headerSize
+
+                # load the taskDef, this method call will be overloaded by
+                # subclasses.
+                # bytes are compressed, so decompress them
+                taskDef = pickle.loads(lzma.decompress(_readBytes(start, stop)))
+                loadedTaskDef[nodeTask] = taskDef
+            # Explicitly overload the "frozen-ness" of nodes to attach the
+            # taskDef back into the un-persisted node
+            object.__setattr__(qNode, 'taskDef', loadedTaskDef[nodeTask])
+            quanta[qNode.taskDef].add(qNode.quantum)
+
+            # record the node for later processing
+            quantumToNodeId[qNode.quantum] = qNode.nodeId
+
+        # Get the id of this graph
+        _buildId = qNode.nodeId.buildId if nodes else None  # type: ignore
+
+        # construct an empty new QuantumGraph object, and run the associated
+        # creation method with the un-persisted data
+        qGraph = object.__new__(QuantumGraph)
+        qGraph._buildGraphs(quanta, _quantumToNodeId=quantumToNodeId, _buildId=_buildId)
+        return qGraph
+
+    def _readBytes(self, start: int, stop: int) -> bytes:
+        """Loads the specified byte range from the ButlerURI object
+
+        In the base class, this actually will read all the bytes into a buffer
+        from the specified ButlerURI object. Then for each method call will
+        return the requested byte range. This is the most flexible
+        implementation, as no special read is required. This will not give a
+        speed up with any sub graph reads though.
+        """
+        if not hasattr(self, 'buffer'):
+            self.buffer = self.uriObject.read()
+        return self.buffer[start:stop]
+
+    def close(self):
+        """Cleans up an instance if needed. Base class does nothing
+        """
+        pass
+
+
+@register_helper(ButlerS3URI)
+class S3LoadHelper(DefaultLoadHelper):
+    # This subclass implements partial loading of a graph using a s3 uri
+    def _readBytes(self, start: int, stop: int) -> bytes:
+        args = {}
+        # minus 1 in the stop range, because this header is inclusive rather
+        # than standard python where the end point is generally exclusive
+        args["Range"] = f"bytes={start}-{stop-1}"
+        try:
+            response = self.uriObject.client.get_object(Bucket=self.uriObject.netloc,
+                                                        Key=self.uriObject.relativeToPathRoot,
+                                                        **args)
+        except (self.uriObject.client.exceptions.NoSuchKey,
+                self.uriObject.client.exceptions.NoSuchBucket) as err:
+            raise FileNotFoundError(f"No such resource: {self}") from err
```

Comment:
Do you want `{self.uriObject}` instead of `{self}`?

Reply:
Yep, I didnt catch that when I was looking at the daf_butler implementation.